src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests.cs

```diff
@@ -9202,6 +9413,414 @@ public UnmanagedCallersOnlyAttribute()
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion9, "C<ERROR>").WithArguments("generic attributes", "10.0").WithLocation(4, 2),
+                // (4,4): error CS0246: The type or namespace name 'ERROR' could not be found (are you missing a using directive or an assembly reference?)
+                // [C<ERROR>]
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "ERROR").WithArguments("ERROR").WithLocation(4, 4),
+                // (5,2): error CS8773: Feature 'generic attributes' is not available in C# 9.0. Please use language version 10.0 or greater.
+                // [C<System>]
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion9, "C<System>").WithArguments("generic attributes", "10.0").WithLocation(5, 2),
+                // (5,4): error CS0118: 'System' is a namespace but is used like a type
+                // [C<System>]
+                Diagnostic(ErrorCode.ERR_BadSKknown, "System").WithArguments("System", "namespace", "type").WithLocation(5, 4),
+                // (6,2): error CS7003: Unexpected use of an unbound generic name
+                // [C<>]
+                Diagnostic(ErrorCode.ERR_UnexpectedUnboundGenericName, "C<>").WithLocation(6, 2),
+                // (6,2): error CS8773: Feature 'generic attributes' is not available in C# 9.0. Please use language version 10.0 or greater.
+                // [C<>]
+                Diagnostic(ErrorCode.ERR_FeatureNotAvailableInVersion9, "C<>").WithArguments("generic attributes", "10.0").WithLocation(6, 2));
+
+            comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (4,4): error CS0246: The type or namespace name 'ERROR' could not be found (are you missing a using directive or an assembly reference?)
+                // [C<ERROR>]
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "ERROR").WithArguments("ERROR").WithLocation(4, 4),
+                // (5,4): error CS0118: 'System' is a namespace but is used like a type
+                // [C<System>]
+                Diagnostic(ErrorCode.ERR_BadSKknown, "System").WithArguments("System", "namespace", "type").WithLocation(5, 4),
+                // (6,2): error CS7003: Unexpected use of an unbound generic name
+                // [C<>]
+                Diagnostic(ErrorCode.ERR_UnexpectedUnboundGenericName, "C<>").WithLocation(6, 2));
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
+        public void GenericAttribute_Reflection_CoreClr()
+        {
+            var source = @"
+using System;
+
+public class Attr<T> : Attribute { }
+
+[Attr<int>]
+public class Program {
+    public static void Main() {
+        var attr = Attribute.GetCustomAttribute(typeof(Program), typeof(Attr<int>));
+        Console.Write(attr);
+    }
+}
+";
+            var verifier = CompileAndVerify(source, expectedOutput: "Attr`1[System.Int32]");
+            verifier.VerifyDiagnostics();
+        }
+
         #endregion
```

Comment:
>endregion [](http://example.com/codeflow?start=9&length=9)

Consider testing `AllowMultiple = false`.
```C#
using System;
[AttributeUsage(AllowMultiple = false)]
class MyAttribute<T> : Attribute
{
}
[MyAttribute<int>]
[MyAttribute<object>] // ok?
[MyAttribute<int>] // error?
class Program
{
}
```

Reply:
Why would the second `[MyAttribute<int>]` be an error?